git_perf/src/config.rs

```diff
@@ -245,149 +419,484 @@ epoch = "{}"
+audit_min_relative_deviation = 1.0
+"#,
+            )
+            .unwrap();
+
+            // Create workspace config that overrides some values
+            let workspace_config_path = temp_dir.join(".gitperfconfig");
+            fs::write(
+                &workspace_config_path,
+                r#"
+[measurement."test"]
+backoff_max_elapsed_seconds = 60
+"#,
+            )
+            .unwrap();
+
+            // Set HOME to our temp directory
+            env::set_var("HOME", temp_dir);
+            env::remove_var("XDG_CONFIG_HOME");
+
+            // Read hierarchical config and verify workspace overrides home
+            let config = read_hierarchical_config().unwrap();
+
+            // backoff_max_elapsed_seconds should be overridden by workspace config
+            let backoff: i32 = config
+                .get("measurement.test.backoff_max_elapsed_seconds")
+                .unwrap();
+            assert_eq!(backoff, 60);
+
+            // audit_min_relative_deviation should come from home config
+            let deviation: f64 = config
+                .get("measurement.test.audit_min_relative_deviation")
+                .unwrap();
+            assert_eq!(deviation, 1.0);
+        });
+    }
+
+    #[test]
+    fn test_determine_epoch_from_config_with_missing_file() {
+        // Test that missing config file doesn't panic and returns None
+        let temp_dir = TempDir::new().unwrap();
+        fs::create_dir_all(temp_dir.path()).unwrap();
+        env::set_current_dir(temp_dir.path()).unwrap();
+
+        let epoch = determine_epoch_from_config("test_measurement");
+        assert!(epoch.is_none());
+    }
 
-        // Case 5: invalid config (should return StandardDeviation)
-        let configfile = "[audit]\ndispersion_method = invalid\n";
+    #[test]
+    fn test_determine_epoch_from_config_with_invalid_toml() {
+        let temp_dir = TempDir::new().unwrap();
+        let config_path = temp_dir.path().join(".gitperfconfig");
+        fs::write(&config_path, "invalid toml content").unwrap();
+
+        fs::create_dir_all(temp_dir.path()).unwrap();
+        env::set_current_dir(temp_dir.path()).unwrap();
+
+        let epoch = determine_epoch_from_config("test_measurement");
+        assert!(epoch.is_none());
+    }
+
+    #[test]
+    fn test_write_config_creates_directories() {
+        let temp_dir = TempDir::new().unwrap();
+        let nested_dir = temp_dir.path().join("a").join("b").join("c");
+        let config_path = nested_dir.join(".gitperfconfig");
+
+        // Store original directory before any changes
+        let original_dir = match env::current_dir() {
+            Ok(dir) => dir,
+            Err(_) => {
+                // If current directory is invalid, use a safe fallback
+                std::env::var("HOME")
+                    .map(std::path::PathBuf::from)
+                    .unwrap_or_else(|_| "/tmp".into())
+            }
+        };
+
+        fs::create_dir_all(&nested_dir).unwrap();
```

Comment:
@cursoragent this is not a good test anymore. Remove it as we no longer have nested config files.